cql3/statements/select_statement.cc

```diff
@@ -379,6 +382,29 @@ select_statement::do_execute(service::storage_proxy& proxy,
     command->slice.options.set<query::partition_slice::option::allow_short_read>();
     auto timeout_duration = get_timeout(state.get_client_state(), options);
     auto timeout = db::timeout_clock::now() + timeout_duration;
+
+    if (aggregate && _selection->is_count()) {
+        query::forward_request req = {
```

Comment:
I'd like to detect this at prepare() time and create a different select_statement subclass instead, similar to how we have indexed_table_select_statement. This won't have a huge impact, but that's the direction the code should be moving in - making more decisions at prepare time and fewer during runtime.